lib/chef/policy_builder/policyfile.rb

```diff
@@ -142,6 +132,9 @@ def build_node
         expand_run_list
         apply_policyfile_attributes
 
+        if node.run_list && Chef::Config[:policy_persist_run_list]
```

Comment:
If this isn't set, but you have node.run_list shouldn't we fail with a helpful message?

Reply:
no, the node.run_list can be set on people's node objects now while they're using policyfiles and it'll just be ignored.  that would probably be breaking behavior.  i'm not sure if it happens commonly with always-native-policyfile workflows, but people that have converted to policyfiles would certainly have nodes that have node.run_lists set which are currently being ignored.  that is one reason why the config option HAS to be a config option and it CANNOT be enabled by default (i briefly toyed with that idea yesterday, but its probably better making it opt-in anyway even if the breaking change wasn't an issue).